Source/WebCore/page/EventHandler.cpp

```diff
@@ -3295,6 +3295,15 @@ bool EventHandler::sendContextMenuEvent(const PlatformMouseEvent& event)
 {
     Ref protectedFrame(m_frame);
 
+#if ENABLE(POINTER_LOCK)
+    // Context menus should not be handled while pointer is locked.
+    auto* page = m_frame.page();
+    if (!page)
+        return false;
```

Comment:
Nit - I think you could also consider moving these three lines out of the `POINTER_LOCK` compile-time flag.

Reply:
Yeah that's fair, though that would go against my amendment for a better scoped `page`. Thoughts?

```cpp
#if ENABLE(POINTER_LOCK)
if (auto* page = m_frame.page(); !page || page->pointerLockController().isLocked())
    return false;
#endif
```

Reply:
The latter has the (slight) advantage that it keeps the `page` nullptr early return consistent, regardless of `POINTER_LOCK` enablement, but at the cost of making the style a bit weird. A third option is to wrap this in a `{ }` so that the raw `Page*` pointer won't live outside the scope of the check. TBH, any of the above sound okay to me.